src/components/VmDetails/cards/DetailsCard/index.js

```diff
@@ -110,6 +110,26 @@ function createTemplateList (templates) {
   return templateList
 }
 
+/*
+ * Return a normalized and sorted list of OS ready for use in a __SelectBox__ from
+ * the Map of provided templates.
+ */
+function createOsList (architecture) {
```

Comment:
I suggest pushing the `clusterId`, `cluster` and `architecture` lookups to this function and just pass in the VM and Clusters. That way those lookups only happen here instead of being repeated in the constructor and `componentDidUpdate`.